packages/flutter/lib/src/services/text_layout_metrics.dart

```diff
@@ -54,6 +54,23 @@ abstract class TextLayoutMetrics {
     return true;
   }
 
+  /// Check if the given code unit is a line terminator character.
+  ///
+  /// Includes newline characters from ASCII
+  /// (https://www.unicode.org/standard/reports/tr13/tr13-5.html).
+  static bool isLineTerminator(int codeUnit) {
```

Comment:
Looking at `TextPainter`, the list of what's considered a newline is different: no 0xD on its own, plus 0x2028/0x2029.

```
  static bool _isNewline(int codePoint) {
    switch (codePoint) {
      case 0x000A:
      case 0x0085:
      case 0x000B:
      case 0x000C:
      case 0x2028:
      case 0x2029:
        return true;
      default:
        return false;
    }
  }
```

Reply:
Hmm I got the list from here https://www.unicode.org/standard/reports/tr13/tr13-5.html. I added `0x2028` and `0x2029` I had left out.